### PR TITLE
UMass: Optimization and Namespace Implementation

### PR DESCRIPTION
**Pull request by JeremyTubongbanua**
**What I did**
- Followed Micropython optimization tips from [this video by the creator of Micropython](https://www.youtube.com/watch?v=hHec4qL00x0&t=651s&ab_channel=PyConAU)
- Implemented namespaces for `put_public` and `get_public`

**Optimization Notes**
Some notes I took from the video:

_Memory Allocation_
- use expressions (long lines instead of splitting it up into variables) << goes against programming principles, but in this case, makes it faster
- importing modules uses a lot of heap, better to import the functions instead of the entire module
- avoid global variables

_CPU_
- use functions, not global scope
- use local variables
- cache the module functions and object methods as locals
- cache self variables as locals (still need to do)*

_RAM_
- shorter variable names are better
- temporary buffers e.g. `x = bytearray(1)` instead of creating it later on is better

**Screenshots from the Video**
image

image

image

### CODE REVIEW

File: lib/at_client/at_client.py [hunk 1]
```diff
@@ -100,33 +100,42 @@ def put_public(self, keyName: str, value: str, ttr: int = 200) -> str:
...
-        import time
+        from utime import sleep
 
         # get the public key
-        verb = 'plookup:bypassCache:true:%s%s' %(keyName, otherAtSign)
-        time.sleep(1)
+        fullKeyName = keyName
+        if not namespace == None:
+            fullKeyName = keyName + '.' + namespace
+        verb = 'plookup:bypassCache:true:%s%s' %(fullKeyName, otherAtSign)
```
Comment (JeremyTubongbanua) [inline lib/at_client/at_client.py:132]: @TylerTrott is this namespace implementation fine?

File: lib/at_client/at_client.py [hunk 2]
```diff
@@ -100,33 +100,42 @@ def put_public(self, keyName: str, value: str, ttr: int = 200) -> str:
...
+        
         from lib.at_client.at_utils import format_atSign
-        verb = 'update:public:%s%s %s' %(keyName, format_atSign(self.atSign), value)
-        del format_atSign
-        import time
-        time.sleep(1)
+        fullKeyName = keyName
+        if not namespace == None:
+            fullKeyName = keyName + '.' + namespace
+        verb = 'update:public:%s%s %s' %(fullKeyName, format_atSign(self.atSign), value)
```
Comment (JeremyTubongbanua) [inline lib/at_client/at_client.py:109]: @TylerTrott

Comment (TylerTrott) [reply lib/at_client/at_client.py:109]: lgtm

File: lib/at_client/at_client.py (hunk 1 above)
Comment (TylerTrott) [reply lib/at_client/at_client.py:132]: yeah, it's how I did it 👍 Nice work

**Review by TylerTrott — APPROVED** [review]
lgtm

**Pull request merged**